examples/jsm/shaders/SSShadowShader.js

```diff
@@ -0,0 +1,279 @@
+			vec3 viewPosition=getViewPosition( vUv, depth, clipW );
+
+			vec2 d0=gl_FragCoord.xy;
+			vec2 d1;
+
+			vec3 viewNormal=getViewNormal( vUv );
+
+			vec3 viewLightPosition=(cameraMatrixWorldInverse*vec4(lightPosition,1.)).xyz;
+			vec3 viewRefractDir=normalize(viewLightPosition-viewPosition);
+
+			float maxDistance2=length(viewPosition-viewLightPosition);
+			maxDistance2=min(maxDistance,maxDistance2);
+			vec3 d1viewPosition=viewPosition+viewRefractDir*maxDistance2;
+			#ifdef PERSPECTIVE_CAMERA
+				if(d1viewPosition.z>-cameraNear){
+					//https://tutorial.math.lamar.edu/Classes/CalcIII/EqnsOfLines.aspx
+					float t=(-cameraNear-viewPosition.z)/viewRefractDir.z;
+					d1viewPosition=viewPosition+viewRefractDir*t;
+				}
+			#endif
+			d1=viewPositionToXY(d1viewPosition);
+
+			float totalLen=length(d1-d0);
+			float xLen=d1.x-d0.x;
+			float yLen=d1.y-d0.y;
+			float totalStep=max(abs(xLen),abs(yLen));
+			float xSpan=xLen/totalStep;
+			float ySpan=yLen/totalStep;
+
+			gl_FragColor=texture2D(tDiffuse,vUv);
+			float attenuation=max(0.,min(1.,1.-length(viewPosition-viewLightPosition)/maxDistance));
+			// float attenuation=max(0.,min(1.,1.-length(viewPosition-viewLightPosition)));
+			attenuation*=attenuation;
+			gl_FragColor.rgb*=attenuation;
+
+			for(float i=0.;i<float(MAX_STEP);i++){
+				if(i>=totalStep) break;
+				vec2 xy=vec2(d0.x+i*xSpan,d0.y+i*ySpan);
+				if(xy.x<0.||xy.x>resolution.x||xy.y<0.||xy.y>resolution.y) break;
+				float s=length(xy-d0)/totalLen;
+				vec2 uv=xy/resolution;
+
+				float d = getDepth(uv);
+				float vZ = getViewZ( d );
+				float cW = cameraProjectionMatrix[2][3] * vZ+cameraProjectionMatrix[3][3];
+				vec3 vP=getViewPosition( uv, d, cW );
+
+				#ifdef PERSPECTIVE_CAMERA
+					// https://www.comp.nus.edu.sg/~lowkl/publications/lowk_persp_interp_techrep.pdf
+					float recipVPZ=1./viewPosition.z;
+					float viewRefractRayZ=1./(recipVPZ+s*(1./d1viewPosition.z-recipVPZ));
+					float sD=surfDist*cW;
+				#else
+					float viewRefractRayZ=viewPosition.z+s*(d1viewPosition.z-viewPosition.z);
+					float sD=surfDist;
+				#endif
+
+				bool hit;
+				#ifdef INFINITE_THICK
+					hit=viewRefractRayZ<=vZ;
+				#else
+					if(viewRefractRayZ-sD>vZ) continue;
+					float away=pointToLineDistance(vP,viewPosition,d1viewPosition);
+					hit=away<=sD;
+				#endif
+				if(hit){
+					vec3 vN=getViewNormal( uv );
+
+					// if(dot(viewRefractDir,vN)>=0.) continue;
+
+					if((length(viewPosition-vP)<doubleSideCheckStartFrom)&&(dot(viewRefractDir,vN)>=0.)) continue;
+					// May not need "doubleSideCheckStartFrom", use "surfDist" or change starting "i" of "for(float i=1.;i<float(MAX_STEP);i++){" instead.
+
+					// gl_FragColor=texture2D(tDiffuse,vUv);
+					gl_FragColor.rgb*=.5;
+					return;
+				}
+			} // end of for loop
+
+			gl_FragColor.rgb*=dot(viewNormal,viewRefractDir)*.5+.5;
+
+		}
+	`
+
+};
+
+var SSShadowDepthShader = {
```

Comment:
Why is this declared using 'var'?

Reply:
Copy-pasted from old SSRShader.js, should be ok with `let` `const`.